src/p5xr/p5vr/p5vr.js

```diff
@@ -49,46 +46,67 @@ export default class p5vr extends p5xr {
       window.setup();
       p5.instance._millisStart = window.performance.now();
     }
+    const refSpaceRequest = this.isImmersive ? 'local' : 'viewer';
+      this.xrSession.requestReferenceSpace(refSpaceRequest)
+        .then((refSpace) => {
+          this.xrRefSpace = refSpace;
+          // Inform the session that we're ready to begin drawing.
+          this.xrSession.requestAnimationFrame(this.onXRFrame.bind(this));
+          if (!this.isImmersive) {
+            this.xrSession.updateRenderState({
+              baseLayer: new XRWebGLLayer(this.xrSession, this.gl),
+              inlineVerticalFieldOfView: 90 * (Math.PI / 180),
+            });
+            this.addInlineViewListeners(this.canvas);
+          }
+        });
     this.onRequestSession();
   }
 
   /**
-   * `device.requestSession()` must be called within a user gesture event.
+   * `navigator.xr.requestSession('immersive-vr')` must be called within a user gesture event.
    * @param {XRDevice}
    */
   onXRButtonClicked() {
-    if (this.isImmersive) {
-      console.log('requesting session with mode: immersive-vr');
+    if (this.hasImmersive) {
+      console.log('Requesting session with mode: immersive-vr');
+      this.isImmersive = true; 
+      this.xrDevice = null;
+      this.xrSession = null;
+      this.xrRefSpace = null;
+      this.xrViewerSpace = null;
+      this.xrHitTestSource = null;
+      this.gl = null;
+      this.frame = null;
```

Comment:
Perhaps break these lines out into a separate function so that the purpose can be made clear by the function name.

Reply:
Moved to separate function: 87a18fa141e71f5f887967fe31fe5dfebe79f497